sentry/src/routes/campaign.rs

```diff
@@ -1,56 +1,286 @@
+        redis::cmd("INCRBY")
+            .arg(&key)
+            .arg(amount.to_u64())
+            .query_async(&mut redis.clone())
+            .await?;
+        Ok(true)
+    }
+
+    pub async fn decrease_remaining_for_campaign(redis: &MultiplexedConnection, id: CampaignId, amount: UnifiedNum) -> Option<UnifiedNum> {
+        let key = format!("{}:{}", CAMPAIGN_REMAINING_KEY, id);
+        let value = match redis::cmd("DECRBY")
+            .arg(&key)
+            .arg(amount.to_u64())
+            .query_async::<_, Option<i64>>(&mut redis.clone())
+            .await
+            {
+                Ok(Some(remaining)) => {
+                    // Can't be less than 0 due to max()
+                    Some(UnifiedNum::from_u64(max(0, remaining).try_into().unwrap()))
+                },
+                _ => None
+            };
+        value
+    }
+
+    pub async fn handle_route<A: Adapter>(
+        req: Request<Body>,
+        app: &Application<A>,
+    ) -> Result<Response<Body>, ResponseError> {
+        let campaign = req.extensions().get::<Campaign>().expect("We must have a campaign in extensions");
+        let modified_campaign = ModifyCampaign::from_campaign(campaign.clone());
+
+        // Getting the campaign as it was before the update operation, should exist
+        let campaign_being_mutated = fetch_campaign(app.pool.clone(), &campaign.id).await?.ok_or(ResponseError::NotFound)?;
+
+        // modify Campaign
+        modify_campaign(&app.pool, &campaign_being_mutated, &modified_campaign, &app.redis).await.map_err(|_| ResponseError::BadRequest("Failed to update campaign".to_string()))?;
+
+        Ok(success_response(serde_json::to_string(&campaign)?))
+    }
+
+    pub async fn modify_campaign(pool: &DbPool, campaign: &Campaign, modified_campaign: &ModifyCampaign, redis: &MultiplexedConnection) -> Result<Campaign, Error> {
+        // *NOTE*: When updating campaigns make sure sum(campaigns.map(getRemaining)) <= totalDepoisted - totalspent
+        // !WARNING!: totalSpent != sum(campaign.map(c => c.spending)) therefore we must always calculate remaining funds based on total_deposit - lastApprovedNewState.spenders[user]
+        // *NOTE*: To close a campaign set campaignBudget to campaignSpent so that spendable == 0
+
+        if let Some(new_budget) = modified_campaign.budget {
+            let old_remaining = get_remaining_for_campaign(&redis, campaign.id).await?.ok_or(Error::FailedUpdate("No remaining entry for campaign".to_string()))?;
+
+            let campaign_spent = campaign.budget.checked_sub(&old_remaining).ok_or(Error::Calculation)?;
+            if campaign_spent >= new_budget {
+                return Err(Error::NewBudget("New budget should be greater than the spent amount".to_string()));
+            }
+
+            // Separate variable for clarity
+            let old_budget = campaign.budget;
+
+            match new_budget.cmp(&old_budget) {
+                Ordering::Greater | Ordering::Equal => {
```

Comment:
if New & Old budgets are equal you can skip this step of calculating and calling redis.